Webapp/sources/blueprints/solvent_PCA/routes.py

```diff
@@ -0,0 +1,396 @@
+import json
+from datetime import date, datetime
+
+import numpy as np
+import pandas as pd
+from flask import Response, jsonify, render_template, request
+from flask_login import login_required
+from sources import models
+from sources.services.person import from_current_user_email
+from sources.services.PCA_graph import list_user_graphs, PCAgraph_from_id
+from sources.auxiliary import get_notification_number, get_workgroups
+from sources.extensions import db
+
+from . import solvent_PCA_bp
+from Webapp.sources.services.solvent_PCA.embedding import get_PCA, get_r_class_descriptors
```

Comment:
same with these imports - I've been sticking to the pattern of importing services and then just using services.domain.function()